gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/CopySource.java

```diff
@@ -291,6 +292,20 @@ private void submitUnfulfilledRequestEventsHelper(List<FileSet<CopyEntity>> file
     }
   }
 
+  void failJobIfAllRequestsRejected(RequestAllocator<FileSet<CopyEntity>> allocator,
+      Iterator<FileSet<CopyEntity>> allocatedRequests) throws IOException {
+    // TODO: we should set job as partial success if there is a mix of allocated requests and rejections
+    if (PriorityIterableBasedRequestAllocator.class.isAssignableFrom(allocator.getClass())) {
+      PriorityIterableBasedRequestAllocator<FileSet<CopyEntity>> priorityIterableBasedRequestAllocator =
+          (PriorityIterableBasedRequestAllocator<FileSet<CopyEntity>>) allocator;
+      // If there are no allocated items and there items exceeding the available resources, then we can infer all items exceed resources
```

Comment:
If there are no allocated items and there "are" items exceeding the available resources, then we can infer all items exceed resources?